bevy_app doesn't compile without default features

## Bevy version

latest as of writing ([b0409f6](https://github.com/bevyengine/bevy/commit/b0409f63d5488b209d80f75aa332b6515c8b80d3))

## What you did

```toml
[dependencies]
bevy_app = { path = "../bevy_app", version = "0.14.0-dev", default-features = false }
```

## What went wrong

```
   Compiling bevy_app v0.14.0-dev (F:\Code\bevy\crates\bevy_app)
error[E0425]: cannot find value `StateTransition` in this scope
   --> crates\bevy_app\src\main_schedule.rs:178:17
    |
178 |                 StateTransition.intern(),
    |                 ^^^^^^^^^^^^^^^ not found in this scope
```

--- a/crates/bevy_app/src/main_schedule.rs
+++ b/crates/bevy_app/src/main_schedule.rs
@@ -175,6 +175,7 @@ impl Default for MainScheduleOrder {
             labels: vec![
                 First.intern(),
                 PreUpdate.intern(),
+                #[cfg(feature = "bevy_state")]
                 StateTransition.intern(),
                 RunFixedMainLoop.intern(),
                 Update.intern(),

